Sources/Subs-Db-postgresql.php

```diff
@@ -922,4 +923,50 @@ function smf_db_escape_wildcard_string($string, $translate_human_wildcards = fal
 	return strtr($string, $replacements);
 }
 
+/**
+ * Updates data in a table, using data from other tables
+ *
+ * @param array $table Associative array with info about the table to be updated ('name' => '{db_prefix}foo', 'alias' => 'f')
+ * @param array $from Array of associative arrays with info about the other tables to get data from ('name' => '{db_prefix}bar', 'alias' => 'b', 'condition' => 'f.baz = b.qux')
+ * @param string $set A string containing the SET instructions for the update query
+ * @param string $where A string containing any WHERE conditions for the update query
+ * @param array $db_values The values to be inserted into the compiled query string
+ * @param object $connection The connection to use (if null, $db_connection is used)
+ * @return bool True if the update was successful, otherwise false
+ */
+function smf_db_update_from($table, $joined, $set, $where, $db_values, $connection = null)
+{
+	global $smcFunc, $db_connection, $db_prefix;
+
+	$connection = $connection === null ? $db_connection : $connection;
+
+	if (empty($table['name']) || empty($table['alias']) || empty($set))
+		return;
+
+	$from = array();
+	foreach ($joined as $join)
+	{
+		if (empty($join['name']) || empty($join['alias']) || empty($join['condition']))
+			continue;
+
+		$from[] = $join['name'] . ' AS ' . $join['alias'];
+		$where = (!empty($where) ? $where .' AND ' : '') . $join['condition'];
```

Comment:
would use brackets to differ between where and join condition
exp.
$where = "asd = 1 or bsf = 2";
$join['condition'] = "b.id = a.id"

result of the existing solution:
where asd = 1 or bsf = 2 and b.id = a.id -> could lead in a wrong result
with brackets:
where (asd = 1 or bsf = 2) and (b.id = a.id) -> safe